libraries/src/Service/Provider/Http.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ * Joomla! Content Management System
+ *
+ * @copyright   (C) 2024 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Joomla\CMS\Service\Provider;
+
+use Joomla\CMS\Http\HttpFactoryInterface;
+use Joomla\DI\Container;
+use Joomla\DI\ServiceProviderInterface;
+use Joomla\Http\HttpFactory;
+
+// phpcs:disable PSR1.Files.SideEffects
+\defined('_JEXEC') or die;
+// phpcs:enable PSR1.Files.SideEffects
+
+/**
+ * Service provider for the application's PSR-3 logger dependency
+ *
+ * @since  __DEPLOY_VERSION__
+ */
+class Http implements ServiceProviderInterface
+{
+    /**
+     * Registers the service provider with a DI container.
+     *
+     * @param   Container  $container  The DI container.
+     *
+     * @return  void
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    public function register(Container $container)
+    {
+        $container->alias('http', HttpFactoryInterface::class)
```

Comment:
```suggestion
        $container->alias('http.factory', HttpFactoryInterface::class)
```

On other providers, we usually append .factory to the key, like `document.factory`, `menu.factory`..., so I think we should do the same for HttpFactory

Reply:
Done